src/persistence/StorageNode.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Time } from "../time/Time";
+import { Storage } from "./Storage";
+import { readFile, writeFile } from "fs/promises";
+import { Logger } from "../log/Logger";
+import { ByteArray } from "@project-chip/matter.js";
+import {StorageInMemory} from "./StorageInMemory";
+
+/** We store changes 1s after a value was set to the storage, but not more often than every 1s. */
+const COMMIT_DELAY = 1000 /* 1s */;
+
+const logger = Logger.get("StorageNode");
+
+export class StorageNode extends StorageInMemory {
+    private readonly commitTimer = Time.getTimer(COMMIT_DELAY, () => this.commit());
+    private waitForCommit = false;
+    private allowToCommit = false;
+
+    constructor(
+        private readonly path: string,
+    ) {
+        super();
+    }
+
+    async initialize() {
+        if (Object.keys(this.store).length > 0) {
```

Comment:
initialize() should be called before this object is used, so you can simply use a boolean initialized to track this and prevent get set calls.
No need for the allowToCommit as well.

Reply:
The allowToCommit is to prevent changes after "close" is called ... so has nothing really to do with initialization, but yes could be combined. I now combined it into initilized flag ... If you have a better name just tell, because "close" now also resets "initialized" to false

Reply:
Better not to reset the initialized flag.
Usually the flag to track if the object is closed is named "closed".

Note: close() is never called in the current implementation...

Reply:
Ok I added another closed flag now. and close on PeristenceManager